ballerina/tests/in-memory-id-fields-tests.bal

```diff
@@ -0,0 +1,472 @@
+
+    // read one dependent
+    BooleanIdRecordDependent retrievedRecord1Dependent = check testEntitiesClient->/booleanidrecords/[booleanIdRecord1.id].get();
+    test:assertEquals({randomField: booleanIdRecord1.randomField}, retrievedRecord1Dependent);
+
+    // read
+    BooleanIdRecord[] booleanIdRecords = check from BooleanIdRecord booleanIdRecord in testEntitiesClient->/booleanidrecords.get(BooleanIdRecord)
+        select booleanIdRecord;
+    test:assertEquals(booleanIdRecords, [booleanIdRecord1, booleanIdRecord2]);
+
+    // read dependent
+    BooleanIdRecordDependent[] booleanIdRecordsDependent = check from BooleanIdRecordDependent booleanIdRecord in testEntitiesClient->/booleanidrecords.get(BooleanIdRecordDependent)
+        select booleanIdRecord;
+    test:assertEquals(booleanIdRecordsDependent, [{randomField: booleanIdRecord1.randomField}, {randomField: booleanIdRecord2.randomField}]);
+
+    // update
+    retrievedRecord1 = check testEntitiesClient->/booleanidrecords/[booleanIdRecord1.id].put({randomField: booleanIdRecord1Updated.randomField});
+    test:assertEquals(booleanIdRecord1Updated, retrievedRecord1);
+    retrievedRecord1 = check testEntitiesClient->/booleanidrecords/[booleanIdRecord1.id].get();
+    test:assertEquals(booleanIdRecord1Updated, retrievedRecord1);
+
+    // delete
+    BooleanIdRecord retrievedRecord2 = check testEntitiesClient->/booleanidrecords/[booleanIdRecord2.id].delete();
+    test:assertEquals(booleanIdRecord2, retrievedRecord2);
+    booleanIdRecords = check from BooleanIdRecord booleanIdRecord in testEntitiesClient->/booleanidrecords.get(BooleanIdRecord)
+        select booleanIdRecord;
+    test:assertEquals(booleanIdRecords, [booleanIdRecord1Updated]);
+
+    check testEntitiesClient.close();
+}
+
+@test:Config {
+    groups: ["id-fields", "in-memory"]
+}
+function inMemoryAllTypesIdFieldTest() returns error? {
+    InMemoryTestEntitiesClient testEntitiesClient = check new ();
+    AllTypesIdRecord allTypesIdRecord1 = {
+        intType: 1,
+        stringType: "id-1",
+        floatType: 1.0,
+        booleanType: true,
+        decimalType: 1.1d,
+        randomField: "test1"
+    };
+    AllTypesIdRecord allTypesIdRecord2 = {
+        intType: 2,
+        stringType: "id-2",
+        floatType: 2.0,
+        booleanType: false,
+        decimalType: 2.2d,
+        randomField: "test2"
+    };
+    AllTypesIdRecord allTypesIdRecord1Updated = {
+        intType: 1,
+        stringType: "id-1",
+        floatType: 1.0,
+        booleanType: true,
+        decimalType: 1.1d,
+        randomField: "test1Updated"
+    };
+
+    // create
+    [boolean, int, float, decimal, string][] ids = check testEntitiesClient->/alltypesidrecords.post([allTypesIdRecord1, allTypesIdRecord2]);
+    test:assertEquals(ids, [[allTypesIdRecord1.booleanType, allTypesIdRecord1.intType, allTypesIdRecord1.floatType, allTypesIdRecord1.decimalType, allTypesIdRecord1.stringType], 
```

Comment:
Shall we format these files?

Reply:
Fixed in [f525e16](https://github.com/ballerina-platform/module-ballerina-persist/pull/190/commits/f525e164b6461798557a62d1758ce7fa4267032b)